.github/workflows/tests.yml

```diff
@@ -436,21 +438,50 @@ jobs:
 
   complement:
     if: "${{ !failure() && !cancelled() }}"
-    needs: linting-done
+    needs: calculate-test-jobs
     runs-on: ubuntu-latest
 
     strategy:
       fail-fast: false
       matrix:
-        include:
-          - arrangement: monolith
-            database: SQLite
+        job: ${{ fromJson(needs.calculate-test-jobs.outputs.complement_test_matrix) }}
+
+    steps:
+      - name: Run actions/checkout@v3 for synapse
+        uses: actions/checkout@v3
+        with:
+          path: synapse
 
-          - arrangement: monolith
-            database: Postgres
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+            toolchain: 1.58.1
+            override: true
+      - uses: Swatinem/rust-cache@v2
 
-          - arrangement: workers
-            database: Postgres
+      - name: Prepare Complement's Prerequisites
+        run: synapse/.ci/scripts/setup_complement_prerequisites.sh
+
+      - name: Run Complement Tests
+        shell: bash
+        env:
+          POSTGRES: ${{ (matrix.job.database == 'Postgres') && 1 || '' }}
+          WORKERS: ${{ (matrix.job.arrangement == 'workers') && 1 || '' }}
+          WORKER_TYPES: ${{ matrix.job.worker_types }}
+        run: |
+          set -o pipefail
+          COMPLEMENT_DIR=`pwd`/complement synapse/scripts-dev/complement.sh -json 2>&1 | synapse/.ci/scripts/gotestfmt
+
+  back-compat:
```

Comment:
Why wouldn't this be included under the `complement` job above as part of the matrix?

It's also not clear what this is testing backwards compatibility with?